File: 1791-Find-Center-of-Star-Graph.py
import functools

class Solution:
    def findCenter(self, edges: List[List[int]]) -> int:
        return functools.reduce(lambda a, b: b.intersection(a), [set(l) for l in edges]).pop()


# Performance Result:
#   Coding Time: 00:09:24
#   Time Complexity: O(E)
#   Space Complexity: O(E)
#   Runtime: 900 ms, faster than 52.82%
#   Memory Usage: 56.8 MB, less than 5.06%

File: 1791-Find-Center-of-Star-Graph.go
func findCenter(edges [][]int) int {
    cnts := make(map[int]int)
    maxVal := 0
    maxNode := 0
    for _, edge := range edges {
        cnts[edge[0]]++
        if cnts[edge[0]] > maxVal {
            maxVal = cnts[edge[0]]
            maxNode = edge[0]
        }

        cnts[edge[1]]++
        if cnts[edge[1]] > maxVal {
            maxVal = cnts[edge[1]]
            maxNode = edge[1]
        }
    }

    return maxNode
}

// Performance Result:
//   Coding Time: 00:03:41
//   Time Complexity: O(|V|)
//   Space Complexity: O(|V|)
//   Runtime: 197 ms, faster than 5.26%
//   Memory Usage: 17.9 MB, less than 28.95%
